crates/llms/tests/llms/mod.rs

```diff
@@ -196,24 +190,68 @@ async fn run_test_case(
     Ok(())
 }
 
-#[tokio::test]
-#[allow(clippy::expect_used, clippy::expect_fun_call)]
-async fn run_all_tests() {
-    // Set ENV variables before we lazy load `TEST_MODELS`.
-    let _ = dotenvy::from_filename(".env").expect("failed to load .env file");
+async fn run_tests_for_model<F, Fut>(model_name: &str, create_model: F) -> Result<(), anyhow::Error>
+where
+    F: FnOnce() -> Fut,
+    Fut: Future<Output = Result<Arc<dyn Chat>, anyhow::Error>>,
+{
+    // Load environment variables
+    dotenvy::from_filename(".env")?;
+
+    let model = create_model().await?;
 
     for ts in TEST_CASES.iter() {
-        for (model_name, model) in TEST_MODELS.iter() {
-            if crate::llms::TEST_DENY_LIST
-                .iter()
-                .any(|(m, t)| m == model_name && *t == ts.name)
-            {
-                tracing::info!("Skipping test {model_name}/{}", ts.name);
-                continue;
-            }
-            run_test_case(ts, model_name, Arc::clone(model))
-                .await
-                .expect(format!("Failed to run test {model_name}/{}", ts.name).as_str());
+        if crate::llms::TEST_DENY_LIST
+            .iter()
+            .any(|(m, t)| *m == model_name && *t == ts.name)
+        {
+            tracing::info!("Skipping test {model_name}/{}", ts.name);
+            continue;
         }
+        run_test_case(ts, model_name, Arc::clone(&model)).await?;
     }
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_anthropic_model() {
```

Comment:
Why do we need a test per model?

Reply:
Models can be filtered out before the model is loaded, see: https://github.com/spiceai/spiceai/pull/3690/files#diff-4162e2a84adbf9bc3bf9d99f2c9a12fdf04b440211a4dd46a02d0bd48936ae9cR155

Reply:
I prefer current approach as
1. it follows standard approach to run/skip tests
2. better DX as you can run/debug specific model tests directly from IDE  w/o configuring additional env variables

Reply:
Single test will work better if we don't want tests to run in parallel - for example, can local model test and HF test potential conflict?